addons/event/views/event_event_views.xml

```diff
@@ -66,6 +67,8 @@
                             <field name="date_begin" string="Date" widget="daterange" options="{'end_date_field': 'date_end'}" />
                             <field name="date_end" invisible="1" />
                             <field name="date_tz"/>
+                            <field name="date_begin_located" string="Start Date in Timezone" invisible="not display_date_located"/>
+                            <field name="date_end_located" string="End Date in Timezone" invisible="not display_date_located"/>
```

Comment:
You can actually just do `date_tz == context.tz`

Reply:
Hello @reth-odoo !

Thanks for the review, and your suggestion. I have updated the diff to only use the context in the invisible condition.

While I know that there is a paragraph about xml modification in the stable policy:
![stable_policy](https://github.com/user-attachments/assets/55f0047b-607b-4fe9-a2c3-a5b834f354bd)
To me and since the doc is explicitely describing the timezone field differently than the way it is actually used:
![description](https://github.com/user-attachments/assets/07a158dc-6017-4b76-9eec-8fbcceca0cd5)

I do believe that displaying the relevant date somewhere is an inevitable change. In addition, in our case, we simply add 2 already **existing** fields in the view. This does not change any flow by more than a yota and I don't really think that we risk to break any custo/sudio xpath here. The only downside  I see of this chagne in stable is that you don't "-u" you will not update the view and hence won't see the IMP.

Reply:
> doc is explicitly describing the timezone field differently

Not really, it is the timezone of the event. If you change the timezone it does update the time the notifications are sent, the time displayed on the agenda, ...
The bug is more that the time fields don't show the correct time, relative to the event.

> I do believe that displaying the relevant date somewhere is an inevitable change

Yes I think it would be nice to make that more clear, but it's been this way for quite a long time now. I don't really agree that it *needs* to be a stable fix, especially as we've already had this feedback and haven't modified it until now.

IF we *really* want it here:
Currently the fields don't fit well into the form view, and ideally we'd probably want them inlined.
Alternatively we can consider renaming the field to be more clear about what it represents, which is less invasive.

Overall would still say we'll fix in master, will talk about it with the PO of the app when she's back.

Reply:
![description](https://github.com/user-attachments/assets/895a0e3a-9a6e-4d52-8845-0995758ea16c)
This is the description of the fields in the product form, and therefore, in my opinion, it should describe what each field represents for the record on that form.

The timezone used for any conversion of the dates present on the form to the database is not the timezone field. This means the timezone field does not represent the timezone of the event on the form. Instead, it represents the timezone used for the marketing of the event.

If you were to change the timezone field, the event would still target the same agenda slot. Whether it is displayed as 10 AM UTC+0 or 11 AM UTC+1 does not alter the event it represents nor its placement in anyone's agenda.

The fact that both fields' descriptions use the same subject is therefore misleading. A more appropriate description of these fields would be the following:
- **Date**: When the event is scheduled to take placed (expressed in your timezone)
- **Timezone**: The timezone used for the marketing of the event (e.g. public display on the website,...)

I am fine waiting for the PO's opinion on that one.

Reply:
Ok, I see you point. Imo it should be more about updating the doc and renaming the field but we'll see, forwarded to the PO via DM.

Reply:
@lase-odoo PO is also of the opinion the fields add too much noise, but we agreed on renaming the field to "Display Timezone" and specifying in the tooltip "Date will be converted and displayed in this timezone on the website". We can also update the doc to match.

There are active discussions about how to fix this in master, but as we're already trying to tidy the form view for this model adding we'd rather not add fields. "date in timezone" also doesn't help that much with understanding what the original field does as it can also be interpreted as "this is the actual timezone once everything is computed".

So would go with the field string/tooltip update for now (maybe in 16 then?)

Let me know if that works for you, cheers

Reply:
k, will do ! Thanks for the update :slightly_smiling_face: